### DISCUSSION TITLE
How to align bits from DrawMorganBits

### DISCUSSION THREAD

**Discussion by Laurianne50** [rdkit/rdkit#4800] [Q&A]
Hi all,

I am trying to draw specific Morgan Fingerprints Bits considering several molecules. The bits are not shared by all the molecules in my set. Thus when I draw them, they are not aligned because they are drawn considering the position in space they had in the molecules.

Here is a snapshot of the current resulting figure: 
![draw_bits](https://user-images.githubusercontent.com/54038643/146213169-28887c48-319b-4029-b1fe-182332813de3.png)
Is there a way to align the bits atoms and unify their sizes?

Here is my piece of code (running in Jupyter Notebook):

```import pandas as pd
from rdkit import Chem
from rdkit.Chem.Draw import IPythonConsole
from rdkit.Chem import Draw
from rdkit.Chem import AllChem

list_smi = ['Cc1cc(NC(=O)c2ccc(NC(=O)CC(C)C)cc2)no1', 'COc1ccc(N2CC(C(=O)Nc3ccc(S(=O)(=O)Nc4nccs4)cc3)CC2=O)c(OC)c1']

mol_list_rdkit = []
for x in list_smi:
    content = Chem.MolFromSmiles(x)
    mol_list_rdkit.append(content)
bitInfo_list = []
for x in mol_list_rdkit:
    bitInfo_={}
    fp = AllChem.GetMorganFingerprintAsBitVect(x, 2, bitInfo=bitInfo_)
    bitInfo_list.append(bitInfo_)

list_bits = [1,191, 883,1099]

tpls = []
leg = []
for i, molec in enumerate(mol_list_rdkit):
    bits = [x for x in list_bits if x in bitInfo_list[i]]
    leg_list = [str(x) for x in bits]
    leg = leg+leg_list
    tup_list = [(molec,x,bitInfo_list[i]) for x in bits]
    tpls = tpls+tup_list
leg = [int(i) for i in leg]

index_interest = []
for x in list_bits:
    ind = leg.index(x)
    index_interest.append(ind)
leg = [leg[i] for i in index_interest]
leg = [str(i) for i in leg]

tpls = [tpls[i] for i in index_interest]
imma = Draw.DrawMorganBits(tpls,molsPerRow=2,legends=leg)
imma
```
About my configuration :
RDKit version: 2021.03.5
Python version: 3.8.12

Many thanks!

**Comment by greglandrum** [comment]
Hi @Laurianne50 : apologies that we overlooked this one. Unfortunately I don't have a straightforward solution for you.

The bit drawing code always uses the atom positions from the reference molecule(s), so without writing your own version of `DrawMorganBits()`, there's not really a way to do this.

The same thing applies to resizing the bits: `DrawMorganBits()` uses the `DrawMolecules()` methods, and that uses the same scale for all molecules which are drawn at the same time. The only way to resolve this one is to draw the bits individually and then to combine the individual drawings into one output image.
